tests/components/emulated_hue/test_hue_api.py

```diff
@@ -1310,6 +1310,57 @@ async def test_put_then_get_cached_properly(hass, hass_hue, hue_client):
         assert ceiling_json["state"][HUE_API_STATE_BRI] == 127
 
 
+async def test_put_than_get_when_service_call_fails(hass, hass_hue, hue_client):
+    """Test putting and getting the light state when the service call fails."""
+
+    # Turn the bedroom light off first
+    await hass_hue.services.async_call(
+        light.DOMAIN,
+        const.SERVICE_TURN_OFF,
+        {const.ATTR_ENTITY_ID: "light.ceiling_lights"},
+        blocking=True,
+    )
+
+    turn_on_calls = []
+
+    # Now break the turn on service
+    @callback
+    def mock_service_call(call):
+        """Mock service call."""
+        turn_on_calls.append(call)
+
+    hass_hue.services.async_register(
+        light.DOMAIN, SERVICE_TURN_ON, mock_service_call, schema=None
+    )
+
+    ceiling_lights = hass_hue.states.get("light.ceiling_lights")
+    assert ceiling_lights.state == STATE_OFF
+
+    with patch.object(hue_api, "STATE_CHANGE_WAIT_TIMEOUT", 0.000001):
+        await asyncio.sleep(0.000001)
+        # update light state through api
+        await perform_put_light_state(
+            hass_hue,
+            hue_client,
+            "light.ceiling_lights",
+            True,
+            hue=4369,
+            saturation=127,
+            brightness=254,
+        )
+
+    # Ensure we did not actually turn on
+    assert hass.states.get("light.ceiling_lights").state == STATE_OFF
```

Comment:
The first await serves no purpose, it won't affect the put_light_state method as it starts a new timer.

I think this could intermittently fail as there's no guarantee the service won't actually be on (it's all async) with no wait.

Reply:
```
    hass_hue.services.async_register(
        light.DOMAIN, SERVICE_TURN_ON, mock_service_call, schema=None
    )
```

This is mocking out the turn on service call so it is not possible to turn the light on with the service anymore.

Reply:
Oh I see.

I don't see how the await line 1340 makes a difference to anything though.

Reply:
It shouldn't be needed.